prql-python/src/lib.rs

```diff
@@ -69,19 +72,33 @@ pub struct CompileOptions {
     ///
     /// If `None` is used, the `target` argument from the query header is used.
     /// If it does not exist, [Dialect::Generic] is used.
-    pub target: Option<Dialect>,
+    pub target: String,
 
     /// Emits the compiler signature as a comment after generated SQL
     ///
     /// Defaults to true.
     pub signature_comment: bool,
 }
 
+#[pymethods]
+impl CompileOptions {
+    #[new]
+    pub fn new(format: bool, target: String, signature_comment: bool) -> Self {
+        CompileOptions {
+            format,
+            target,
+            signature_comment,
+        }
+    }
+}
+
 impl From<CompileOptions> for prql_compiler::Options {
     fn from(o: CompileOptions) -> Self {
+        let target = Target::from_str(&o.target).unwrap_or(Target::Sql(Some(Dialect::Generic)));
```

Comment:
I think this should fall back to `Target::Sql(None)`, but for now I have it matched to prql-js.

https://github.com/PRQL/prql/blob/6adbba316688be9beb3e3c7a4513b55d4218683c/prql-js/src/lib.rs#L106

Reply:
I definitely agree, particularly if we start raising a warning / error when this differs from what's in the PRQL header.